scala-package/core/src/main/scala/org/apache/mxnet/io/NDArrayIter.scala

```diff
@@ -42,27 +43,40 @@ import scala.collection.immutable.ListMap
 class NDArrayIter(data: IndexedSeq[(String, NDArray)],
                   label: IndexedSeq[(String, NDArray)],
                   private val dataBatchSize: Int, shuffle: Boolean,
-                  lastBatchHandle: String) extends DataIter {
-
+                  lastBatchHandle: String,
+                  dataDType: DType, labelDType: DType,
+                  dataLayout: String, labelLayout: String) extends DataIter {
+
+  // scalastyle:off
+  def this(data: IndexedSeq[NDArray], label: IndexedSeq[NDArray],
+           dataBatchSize: Int, shuffle: Boolean,
+           lastBatchHandle: String,
+           dataName: String, labelName: String,
+           dataDType: DType, labelDType: DType,
+           dataLayout: String, labelLayout: String) {
+    this(IO.initData(data, allowEmpty = false, dataName),
```

Comment:
how about changing/adding API for `initData` that will give you `IndexedSeq[DataDesc]` instead of `IndexedSeq[(String, NDArray)]
`initData` seems to be giving the default names for Inputs and labels.

```
private def initDataDesc(data: Indexed[ND]): Indexed[DataDesc]
```